streamer/src/bidirectional_channel.rs

```diff
@@ -0,0 +1,396 @@
+// this is a service to handle bidirections quinn channel
+use bincode::serialize;
+use crossbeam_channel::{Receiver, Sender};
+use quinn::SendStream;
+use serde_derive::{Deserialize, Serialize};
+use solana_perf::packet::{Packet, PacketBatch};
+use solana_sdk::signature::Signature;
+use std::{
+    collections::HashMap,
+    net::SocketAddr,
+    sync::{Arc, RwLock},
+};
+
+#[derive(Serialize, Deserialize)]
+pub struct QuicReplyMessage {
+    pub transaction_signature: Signature,
+    pub message: String,
+}
+
+pub struct QuicBidirectionalData {
+    pub transaction_signature_map: HashMap<Signature, u64>,
+    pub sender_ids_map: HashMap<u64, Sender<QuicReplyMessage>>,
+    pub sender_socket_address_map: HashMap<SocketAddr, u64>,
+    pub last_id: u64,
+}
+
+#[derive(Clone)]
+pub struct QuicBidirectionalReplyService {
+    data: Arc<RwLock<QuicBidirectionalData>>,
```

Comment:
if this is mainly accessed from tokio runtimes (my assumption, didn't check), it might be better to use a tokio native rwlock:
https://github.com/solana-labs/solana/issues/24644#issuecomment-1378755214

Reply:
moved to tokio::sync::RwLock